### ISSUE TITLE
[1.16.4/Forge] Game Freezes and Crashes when a Ship Assembly is Used

### ISSUE THREAD

**Issue by ultranine** [JSJBDEV/PlatosTransporters#31]
Design Used: 3x5 wooden platform, with four fence posts connecting a 3x5x2 balloon cube

Upon placing the ship assembly and right-clicking, the game froze, then crashed after a few seconds. No crash log was generated.

latest.log: https://gist.github.com/ultranine/f99d09b5b946253122a99a29ab07b6c7
(I'm not sure if this contains anything related to the crash, the game seemed to revert to a previous state upon loading the world a second time.)

This error did not occur upon building a ship consisting of just balloons and an assembly. I believe the crash may have been caused by an issue translating the fence posts.

**Comment by JSJBDEV** [comment]
are you using forge or fabric, and what platos version are you using, also can you confirm that your construction was touching only materials in the blacklist?

**Comment by ultranine** [comment]
I'm using Forge Version 35.1.12, Platos version 1.4.2, and my construction had one block of space between it and the ground.

**Comment by JSJBDEV** [comment]
I will try an recreate this when I can, but if you could get a crash log that would help immensely (or latest.log immediately after it crashes) otherwise I'll try and work it out myself

**Comment by ultranine** [comment]
The crash log never generated due to the unique circumstances, but the latest.log is above. I think the data from before and during the crash was never recorded, but you can check the file to make sure.

**Comment by volkoleg** [comment]
hi, just to append for above from ultranine. i have the same .. 
1) mc 1.16.4 java edition. Network Game Mode (by IP).
2) mc server - java on win10x64 (local Ethernet), multiplayer (2 players).
3) forge-1.16.4-35.1.13.jar
4) Plato's ver 1.4.2 (file: platos-1.4.2.jar)
5) Java x64 Version: 1.8.0_271 (CPUs: 8, JVM Flags: 2 total; -Xmx12288M -Xms12288M)

client (1.16.4) show error: "Connection with Server lost. Internal Exception: java.lang.StackOverflowError"

logs from mc server console:
`Description: Exception in server tick loop

[10:52:03] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
java.lang.NoSuchMethodError: net.minecraft.entity.Entity.func_70016_h(DDD)V
        at gd.rf.acro.platos.network.MoveMessage.lambda$receive$0(MoveMessage.java:44) ~[platos:1.4.2] {re:classloading}
        at net.minecraftforge.fml.network.NetworkEvent$Context.enqueueWork(NetworkEvent.java:215) ~[forge:?] {re:classloading}
        at gd.rf.acro.platos.network.MoveMessage.receive(MoveMessage.java:35) ~[platos:1.4.2] {re:classloading}
        at net.minecraftforge.fml.network.simple.SimpleChannel$MessageBuilder.lambda$consumer$1(SimpleChannel.java:233) ~[forge:?] {re:classloading}
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryDecode$3(IndexedMessageCodec.java:128) ~[forge:?] {re:classloading}
        at java.util.Optional.ifPresent(Unknown Source) ~[?:1.8.0_271] {}
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:128) ~[forge:?] {re:classloading}
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:162) ~[forge:?] {re:classloading}
        at net.minecraftforge.fml.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:80) ~[forge:?] {re:classloading}
        at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-3.0.5-service.jar:?] {}
        at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-3.0.5-service.jar:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:297) ~[eventbus-3.0.5-service.jar:?] {}
        at net.minecraftforge.fml.network.NetworkInstance.dispatch(NetworkInstance.java:84) ~[forge:?] {re:classloading}
        at net.minecraftforge.fml.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:88) ~[forge:?] {re:classloading}
        at java.util.Optional.map(Unknown Source) ~[?:1.8.0_271] {}
        at net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(NetworkHooks.java:88) ~[forge:?] {re:classloading}
        at net.minecraft.network.play.ServerPlayNetHandler.func_147349_a(ServerPlayNetHandler.java:1413) ~[?:?] {re:classloading}
        at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:42) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
        at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:12) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
        at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?] {re:classloading}
        at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) ~[?:?] {re:classloading}
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?] {re:mixin,re:computing_frames,re:classloading}
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:733) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:159) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:716) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:710) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:696) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:645) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:the_bumblezone.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_271] {}
[10:52:03] [Server thread/FATAL] [ne.mi.co.ForgeMod/]: Preparing crash report with UUID da9a6c76-a64e-4da1-bdee-1ee05f52ee8e`

**Comment by JSJBDEV** [comment]
I'm.not sure that is the same error, since they changeling you sent suggests there is an issue with the move packet, but they maybe be linked I will review this when I have time

**Comment by JSJBDEV** [comment]
yeah so the first error in this thread I have tried to recreated but not been able to, im not sure what happened there. ive built a similar ship but it seems to work fine bar having to reload to make it exist which was a known bug, ive fixed the second error

**Comment by CrackerJackBeta** [comment]
I've had a similar bug in the [SFX Toolbox](https://www.modpackindex.com/modpack/23432/sfx-toolbox) modpack via CurseForge. Whenever I place the ship assembly, right-click it, then destroy it, the internal server seems to be frozen. No blocks will drop and trying to save and exit results in an infinite saving screen. The second time I tested this I placed it on grass to see if the valid blocks were causing the issue. Since there is no crash, I have no crash log, so I'm not sure how to help.
